monitoring/queuemon_test.go

```diff
@@ -0,0 +1,186 @@
+	Maxlevel     int  `json:"max_queue_level"`
+	IsFull       bool `json:"queue_is_currently_full"`
+	LimitCount   int  `json:"queue_limit_reached_count"`
+}
+
+// ======= mocked test input queue
+
+// TestMetricsQueue is a test queue for the reporter
+type TestMetricsQueue struct {
+	metricState queue.Metrics
+	limit       uint64
+}
+
+// NewTestQueue returns a new test queue
+func NewTestQueue(limit uint64) *TestMetricsQueue {
+	return &TestMetricsQueue{
+		metricState: queue.Metrics{
+			OldestActiveTimestamp: common.Time(time.Now()),
+			EventCount:            opt.UintWith(0),
+			EventLimit:            opt.UintWith(limit),
+		},
+		limit: limit,
+	}
+}
+
+// BufferConfig doesn't do anything
+func (tq TestMetricsQueue) BufferConfig() queue.BufferConfig {
+	return queue.BufferConfig{}
+}
+
+// Producer doesn't do anything
+func (tq TestMetricsQueue) Producer(_ queue.ProducerConfig) queue.Producer {
+	return nil
+}
+
+// Consumer doesn't do anything
+func (tq TestMetricsQueue) Consumer() queue.Consumer {
+	return nil
+}
+
+// Close Doesn't do anything
+func (tq TestMetricsQueue) Close() error {
+	return nil
+}
+
+// Metrics spoofs the metrics output
+func (tq *TestMetricsQueue) Metrics() (queue.Metrics, error) {
+	//fmt.Printf("Got queue state: %#v\n", tq)
+	tq.metricState.EventCount = opt.UintWith(tq.metricState.EventCount.ValueOr(0) + 1)
+
+	if tq.metricState.EventCount.ValueOr(0) > tq.limit {
+		tq.metricState.EventCount = opt.UintWith(0)
+	}
+
+	return tq.metricState, nil
+}
+
+// ============ mocked output reporter
+
+// simple wrapper to return a generic config object
+func initMonWithconfig(interval int, name string) Config {
+	return Config{
+		Interval: time.Millisecond * time.Duration(interval),
+		Enabled:  true,
+		ExpvarOutput: expvar.Config{
+			Enabled: true,
+			Port:    8081,
+			Host:    "",
+			Name:    name,
+		},
+	}
+
+}
+
+// fetch the expvar data from the http endpoint and return the final queue object to test the metrics outputs
+func fetchExpVars(t *testing.T, client http.Client, endpoint string) expvarQueue {
+	resp, err := client.Get(endpoint) //nolint:noctx //this is a test, with a timeout
+	assert.NoError(t, err)
+	defer resp.Body.Close()
+	assert.Equal(t, resp.StatusCode, 200, "expvar endpoint did not return 200: %#v", resp)
+	httpResp, err := io.ReadAll(resp.Body)
+	assert.NoError(t, err)
+	raw := struct {
+		Memstats runtime.MemStats
+		Cmdline  []string
+		Queue    expvarQueue
+	}{}
+	err = json.Unmarshal(httpResp, &raw)
+
+	assert.NoError(t, err)
+
+	return raw.Queue
+}
+
+// actual tests
+
+func TestSetupMonitor(t *testing.T) {
+	monitor := initMonWithconfig(1, "test")
+	queue := NewTestQueue(10)
+	mon, err := NewFromConfig(monitor, queue)
+	assert.NoError(t, err)
+	mon.Watch()
+	mon.End()
+
+}
+
+func TestReportedEvents(t *testing.T) {
+	monitor := initMonWithconfig(1, "queue")
+
+	var maxEvents uint64 = 10
+	queue := NewTestQueue(maxEvents)
+	mon, err := NewFromConfig(monitor, queue)
+	assert.NoError(t, err)
```

Comment:
[Suggestion]
Use [`require.NoError`](https://pkg.go.dev/github.com/stretchr/testify/require#NoError), it'll abort the test if check fails. This is better for checking required conditions.